docs/oauth.md

```diff
@@ -0,0 +1,169 @@
+# OAuth2 API
+
+OAuth2 is a protocol that allows applications to use Webmaker accounts without needing to store a username and password.
+
+Before you can use OAuth2 you'll need to register your application.
+
+## Registering your application
+
+Webmaker requires the following information to register your application:
+
+* **Name** - The friendly name of your application e.g. "Buckley's Bees".
+* **Description** - A description of your application e.g. "Buckley's Bees sells the best honey in Ontario".
+* **Image** - An image that represents your application.
+* **Homepage URL** - The full URL to your application homepage e.g. "https://buckleysbees.ca".
+* **Redirect URL** - Your application's OAuth2 callback URL e.g. "https://buckleysbees.ca/oauth2/callback".
+
+Once you've registered your application you'll recieve your application credentials:
```

Comment:
This is still missing a bit on *how* or *where* to perform this registration (even if we don't have that implemented yet, let's note where we *intend* to put it and how that would work)